src/tools/singlejar/input_jar.h

```diff
@@ -45,8 +45,6 @@ class InputJar {
 
   ~InputJar() { Close(); }
 
-  int fd() const { return mapped_file_.fd(); }
```

Comment:
Sadly I must ask to you to restore this please. It's used in a piece of internal code.

Reply:
Is it OK if I only restore `int fd()` for non-Windows (with `#ifndef _WIN32`)?

Rationale:
- To get `fd` from `HANDLE`, we need to call `_open_osfhandle` which can fail. After that point, we need to use `_close` instead of `CloseHandle` to clean-up. `Open` and `Close` methods will be a bit too complicated.
- Blaze does not support Windows(?)
- We can't just return dummy `fd` on Windows because that just moves the error from compile-time to run-time if Blaze is ported to Windows in the future.

Reply:
> Is it OK if I only restore `int fd()` for non-Windows (with `#ifndef _WIN32`)?

Yes. Thanks!

> Rationale:
> 
> * To get `fd` from `HANDLE`, we need to call `_open_osfhandle` which can fail. After that point, we need to use `_close` instead of `CloseHandle` to clean-up. `Open` and `Close` methods will be a bit too complicated.
> * Blaze does not support Windows(?)
> * We can't just return dummy `fd` on Windows because that just moves the error from compile-time to run-time if Blaze is ported to Windows in the future.

Please add this rationale to the `#ifdef` site (and please mention that the code is used within Google), so readers will understand the point of the ifdef.

Reply:
Done.